### ISSUE TITLE
Grouped Horizontal bar plot with reduced row spacing

### ISSUE THREAD

**Issue by danhamill** [altair-viz/altair#1947]
For the data:

```
    level_0     level_1  date depth  station      temp      WY    DOY
0       153  1995-11-01     4    in   ASHTON -0.600000  1996.0   32.0
1       618  1996-02-02     4    in   ASHTON -6.100000  1996.0  125.0
2       444  1995-12-29     4    in  REXBURG -0.522222  1996.0   90.0
3      2057  1997-02-17     4    in   ASHTON -0.038889  1997.0  140.0
4      2951  1997-11-14     4    in   ASHTON -0.061111  1998.0   45.0
5      3266  1998-01-16     4    in   ASHTON -0.572222  1998.0  108.0
6      3516  1998-03-07     4    in   ASHTON -0.233333  1998.0  158.0
7      4499  1998-12-20     4    in   ASHTON -0.961111  1999.0   81.0
8      4746  1999-02-08     4    in   ASHTON -0.155556  1999.0  131.0
9      4517  1998-12-24     4    in  REXBURG -3.194444  1999.0   85.0
10     4767  1999-02-12     4    in  REXBURG -2.394444  1999.0  135.0
11     7046  2000-11-13     4    in  REXBURG -0.277778  2001.0   44.0
12     8998  2002-02-11     4    in  REXBURG -0.361111  2002.0  134.0
13    10857  2003-03-04     4    in   ASHTON -0.500000  2003.0  155.0
14    10714  2003-02-08     4    in  REXBURG -1.027778  2003.0  131.0
15    12238  2004-01-04     4    in  REXBURG -0.033333  2004.0   96.0
16    13887  2004-12-27     4    in   ASHTON -0.122222  2005.0   88.0
17    13960  2005-01-08     4    in  REXBURG -0.211111  2005.0  100.0
18    14260  2005-02-27     4    in  REXBURG -1.133333  2005.0  150.0
19    15657  2006-01-02     4    in   ASHTON -0.111111  2006.0   94.0
20    15718  2006-01-12     4    in  REXBURG -0.200000  2006.0  104.0
21    16018  2006-03-03     4    in  REXBURG -0.083333  2006.0  154.0
22    17445  2007-01-11     4    in   ASHTON -0.272222  2007.0  103.0
23    17745  2007-03-02     4    in   ASHTON -0.183333  2007.0  153.0
24    17428  2007-01-08     4    in  REXBURG -0.238889  2007.0  100.0
25    17728  2007-02-27     4    in  REXBURG -0.111111  2007.0  150.0
26    20770  2008-12-18     4    in  REXBURG -0.911111  2009.0   79.0
27    21082  2009-02-08     4    in  REXBURG -0.294444  2009.0  131.0
28    22437  2009-12-07     4    in   ASHTON -6.266667  2010.0   68.0
29    22737  2010-01-26     4    in   ASHTON -0.250000  2010.0  118.0
30    22564  2009-12-28     4    in  REXBURG -0.611111  2010.0   89.0
31    24574  2011-02-12     4    in  REXBURG -0.361111  2011.0  135.0
32    26043  2011-12-30     4    in   ASHTON -0.127778  2012.0   91.0
33    26343  2012-02-18     4    in   ASHTON -0.266667  2012.0  141.0
34    26014  2011-12-25     4    in  REXBURG -3.750000  2012.0   86.0
35    26314  2012-02-13     4    in  REXBURG -0.466667  2012.0  136.0
36    27897  2013-01-19     4    in   ASHTON -0.100000  2013.0  111.0
37    27880  2013-01-16     4    in  REXBURG -0.488889  2013.0  108.0
38    29389  2013-12-10     4    in  REXBURG -0.850000  2014.0   71.0
39    29689  2014-01-29     4    in  REXBURG -2.755556  2014.0  121.0
40    31399  2015-01-25     4    in  REXBURG -0.755556  2015.0  117.0
41    34843  2017-01-21     4    in  REXBURG -0.077778  2017.0  113.0
42    36383  2018-01-18     4    in  REXBURG -0.077778  2018.0  110.0
43    36683  2018-03-09     4    in  REXBURG -0.122222  2018.0  160.0
44    38232  2018-12-11     4    in    RIRIE -0.638889  2019.0   72.0
45    38482  2019-01-30     4    in    RIRIE -0.788889  2019.0  122.0
46    39963  2019-11-26     4    in  REXBURG -0.122222  2020.0   57.0
```

I am trying to create a grouped horizontal bar plot, but to reduce the spacing between rows

```python

chart = alt.Chart(df).mark_bar().encode(
    x=alt.X('DOY:O', axis = alt.Axis(title = "Day of Year (Oct 1 - Sep 30)")),
    y=alt.Y('station', axis = alt.Axis(title=None,ticks=False, labels=False)),
    color=alt.Color('station', legend = alt.Legend(title = None)),
    row = alt.Row('WY'),
)
```

This makes a plot that seems too large to post here. When i change `width` and `height` in `.properties`, the plot is becoming illegable.  Looking for another way, I saw the [`alt.Row`](https://altair-viz.github.io/user_guide/generated/channels/altair.Row.html) documentation says I can specify `spacing` in `alt.Row`, but I pass when I do:

```
chart = alt.Chart(df).mark_bar().encode(
    x=alt.X('DOY:O', axis = alt.Axis(title = "Day of Year (Oct 1 - Sep 30)")),
    y=alt.Y('station', axis = alt.Axis(title=None,ticks=False, labels=False)),
    color=alt.Color('station', legend = alt.Legend(title = None)),
    row = alt.Row('WY', spacing = 5),
)
chart.display()
```

 I get the error:

```
SchemaValidationError: Invalid specification

        altair.vegalite.v3.schema.channels.Row, validating 'additionalProperties'

        Additional properties are not allowed ('spacing' was unexpected)
        

alt.Chart(...)
```

Basically, I want a grouped bar plot that doesn't appear as if it is split into rows.  Is this possible?

**Comment by jakevdp** [comment]
The ``spacing`` argument was added in Altair version 4.0. Update your Altair package, and you should be able to use it.

**Comment by danhamill** [comment]
I am currently using `4.0.0`

```
#
# Name                    Version                   Build  Channel
altair                    4.0.0                    pypi_0    pypi
```

**Comment by jakevdp** [comment]
This generally means that your Jupyter notebook kernel is pointing to a different Python installation than your shell where you run ``conda``. See https://stackoverflow.com/questions/39007571/running-jupyter-with-multiple-python-and-ipython-paths/39022003#39022003 for more information

If you just want to install Altair in the current kernel, you can use Jupyter's ``%pip install`` or ``%conda install`` magic command and it will install the package in the right place.

**Comment by danhamill** [comment]
Do you mean the discrepancy between `__version__` and `conda list` generally comes from a kernel connecting to a different python installation?  

Or does that have to do with Altair not finding the `vegascope` renderer?   

Since I can import `vegascope` and `altair` within the same kernel, I don't think this is the issue.  I am running everything in `base` anaconda.

```
(base) C:\Users\RDCRLDDH>conda list vegascope
# packages in environment at C:\Anaconda3:
#
# Name                    Version                   Build  Channel
vegascope                 1.0.14                   pypi_0    pypi

(base) C:\Users\RDCRLDDH>conda list altair
# packages in environment at C:\Anaconda3:
#
# Name                    Version                   Build  Channel
altair                    4.0.0                    pypi_0    pypi

(base) C:\Users\RDCRLDDH>conda info

     active environment : base
    active env location : C:\Anaconda3
            shell level : 1
       user config file : C:\Users\RDCRLDDH\.condarc
 populated config files : C:\Users\RDCRLDDH\.condarc
          conda version : 4.8.2
    conda-build version : 3.18.8
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.2
       base environment : C:\Anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Anaconda3\pkgs
                          C:\Users\RDCRLDDH\.conda\pkgs
                          C:\Users\RDCRLDDH\AppData\Local\conda\conda\pkgs
       envs directories : C:\Anaconda3\envs
                          C:\Users\RDCRLDDH\.conda\envs
                          C:\Users\RDCRLDDH\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.8.2 requests/2.22.0 CPython/3.7.3 Windows/10 Windows/10.0.17134
          administrator : False
             netrc file : None
           offline mode : False
```

**Comment by jakevdp** [comment]
All of the discrepancies you're seeing are likely due to your Jupyter notebook connecting to a different kernel than you expect it to be connected to.

**Comment by jakevdp** [comment]
When you run ``conda list``, you are using a shell in one environment. When you run ``import altair``, you are using Python in another environment. The two environments are not necessarily the same, even if you run ``jupyter notebook`` from the same shell where you run ``conda list``.. The stackoverflow answer has details that might help.

**Comment by danhamill** [comment]
found vegascope workaround in `4.0.0` in #1939 

Thanks!
